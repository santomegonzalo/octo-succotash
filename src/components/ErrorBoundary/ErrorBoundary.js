import React from "react";

class ErrorBoundary extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      hasError: false
    };
  }

  static getDerivedStateFromError(error) {
    return {
      hasError: true
    };
  }

  render() {
    if (this.state.hasError) {
      return <h1>Oups, something is not working...</h1>;
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
