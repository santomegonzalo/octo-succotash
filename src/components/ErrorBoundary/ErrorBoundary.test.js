import * as React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import ErrorBoundary from './ErrorBoundary';

describe('ErrorBoundary Component', () => {
	function SomethingTest() {
		return <div>this is a test</div>;
	}

	const getWrapperElement = (renderFunc, handleCatchError = jest.fn()) =>
		renderFunc(
			<ErrorBoundary onCatchError={handleCatchError}>
				<SomethingTest />
			</ErrorBoundary>
		);

	it('should render correct UI', () => {
		let wrapper = getWrapperElement(mount);

		expect(toJson(wrapper)).toMatchSnapshot();

		wrapper.setState({ hasError: true });
		wrapper = wrapper.update();

		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should not display anything if there is an error', () => {
		let wrapper = getWrapperElement(mount);

		expect(wrapper.find(SomethingTest)).toHaveLength(1);

		wrapper.setState({ hasError: true });
		wrapper = wrapper.update();

		expect(wrapper.find(SomethingTest)).toHaveLength(0);
	});

	it('should return true on call getDerivedStateFromError', () => {
		expect(ErrorBoundary.getDerivedStateFromError()).toEqual({
			hasError: true
		});
	});
});
