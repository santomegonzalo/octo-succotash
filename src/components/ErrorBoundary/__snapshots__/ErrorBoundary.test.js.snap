// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary Component should render correct UI 1`] = `
<ErrorBoundary
  onCatchError={[MockFunction]}
>
  <SomethingTest>
    <div>
      this is a test
    </div>
  </SomethingTest>
</ErrorBoundary>
`;

exports[`ErrorBoundary Component should render correct UI 2`] = `
<ErrorBoundary
  onCatchError={[MockFunction]}
>
  <h1>
    Oups, something is not working...
  </h1>
</ErrorBoundary>
`;
